### DISCUSSION TITLE
ABI: 2.0.x, 2.1.x

### DISCUSSION THREAD

**Discussion by rwmacleod** [CESNET/libyang#1997] [Q&A] [answered]
Are the 2.0.x, 2.1.x releases intended to be ABI compatible? It would be nice if the VERSION and SOVERSION schemes were documented to save people from detailed commit reviews. Also, there's an ABI check feature that seems to exist in
https://github.com/CESNET/libyang/blob/master/CMakeModules/ABICheck.cmake
but I don't see any docs on how to make use of that. Thanks!

**Comment by michalvasko** [comment, marked as answer]
SOVERSION follows the *semver* versioning scheme as one would expect. However, we do have a separate API version for plugins that changes sometimes and includes NBC changes but it is stored in every plugin so obsolete ones will refused to be loaded. Meaning you should have no unexpected problems when updating *libyang* when keeping the same major SO version, they are API/ABI compatible.

That `ABICheck` module is not documented because it is used mostly internally for checking that we do not accidentally make a NBC change but is also used to generate compatibility reports.
